### ISSUE TITLE
Global.json pinning to not installed version causes load failures and issues in VS

### ISSUE THREAD

**Issue by dasMulli** [dotnet/project-system#3963]
Not sure where to best put this issue.

# Steps to reproduce:

1. Have a .NET Core project + solution (web project in my case) with a global.json pinning the SDK to 2.1.300
2. Have recent VS and only 2.1.400 and 2.1.401 SDKs installed
3. CLI command will then show an error:
```
> dotnet build
A compatible SDK version for global.json version: [2.1.300] from [C:\repos\avs-portal\global.json] was not found
Did you mean to run dotnet SDK commands? Please install dotnet SDK from:
  http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
```
4. Open the solution in VS:

# Expected behaviour

One of:
* The solution loads in VS and the SDK is rolled forward to a more recent installed version
* An error similar to the CLI build warning is displayed, including a link to some documentation which aims to explain to users what has gone wrong and how the problem can be fixed.

# Actual behaviour 

<img width="967" alt="screen shot 2018-09-07 at 10 01 38" src="https://user-images.githubusercontent.com/6129250/45206332-0ccdab80-b285-11e8-8898-f7f67fa866cc.png">

when trying to remove the project from the solution:
<img width="484" alt="screen shot 2018-09-07 at 10 02 07" src="https://user-images.githubusercontent.com/6129250/45206366-1fe07b80-b285-11e8-94d2-e7d26d5a44f7.png">

**Comment by dasMulli** [comment]
maybe cc @livarcocc @KathleenDollard for the SDK experience

**Comment by livarcocc** [comment]
@nguerrera

**Comment by Pilchie** [comment]
Dupe of https://github.com/dotnet/project-system/issues/2467 I think?

**Comment by dasMulli** [comment]
Hm this may be more than not being able to find a specified SDK but rather wenn the resolver can't find an SDK directory. The error message is different (expected imports are missing).
Please close if it is a duplicate.

**Comment by dasMulli** [comment]
My biggest concern is that the error message is super generic and not helpful at all:
```
....csproj : error  : Project file is incomplete. Expected imports are missing.

....csproj : error  : Project file is incomplete. Expected imports are missing.

....csproj : error  : Project file is incomplete. Expected imports are missing.
```

**Comment by dasMulli** [comment]
I knew what places to look for, I'm sure other uses won't.

**Comment by nguerrera** [comment]
I believe it is a dupe of #2467  because #3621 was closed as a dupe of #2461 and is basically the same as this. But note that @davkean said to look at #3621 as well when fixing #2467.

At some point, CPS started handling failed imports differently and allowing projects to load with them. When the failed import is what would have provided common targets / design time targets, then you get this cryptic "Project file is incomplete" messages. 

So I think the symptoms have changed, but it's basically the same issue.

**Comment by Pilchie** [comment]
Closing as a dupe of #2467.
